File: src/test/java/no/nav/fo/veilarbregistrering/arbeidssoker/ArbeidssokerperioderTestdataBuilder.java
package no.nav.fo.veilarbregistrering.arbeidssoker;

import java.util.ArrayList;
import java.util.List;

public class ArbeidssokerperioderTestdataBuilder {

    private List<Arbeidssokerperiode> arbeidssokerperioder;

    public static ArbeidssokerperioderTestdataBuilder arbeidssokerperioder() {
        return new ArbeidssokerperioderTestdataBuilder();
    }

    private ArbeidssokerperioderTestdataBuilder() {
        this.arbeidssokerperioder = new ArrayList<>();
    }

    public ArbeidssokerperioderTestdataBuilder periode(ArbeidssokerperiodeTestdataBuilder arbeidssokerperiode) {
        this.arbeidssokerperioder.add(arbeidssokerperiode.build());
        return this;
    }

    public Arbeidssokerperioder build() {
        return new Arbeidssokerperioder(arbeidssokerperioder);
    }
}

File: src/test/kotlin/no/nav/fo/veilarbregistrering/arbeidssoker/ArbeidssokerperioderTestdataBuilder.kt
package no.nav.fo.veilarbregistrering.arbeidssoker

class ArbeidssokerperioderTestdataBuilder private constructor() {
    private val arbeidssokerperioder: MutableList<Formidlingsgruppeperiode>
    fun periode(arbeidssokerperiode: ArbeidssokerperiodeTestdataBuilder): ArbeidssokerperioderTestdataBuilder {
        arbeidssokerperioder.add(arbeidssokerperiode.build())
        return this
    }

    fun build(): Arbeidssokerperioder {
        return Arbeidssokerperioder(arbeidssokerperioder)
    }

    fun arbeidssokerperiode(formidlingsgruppeperiode: Builder<Formidlingsgruppeperiode>): ArbeidssokerperioderTestdataBuilder {
        arbeidssokerperioder.add(formidlingsgruppeperiode.build())
        return this
    }

    companion object {
        fun arbeidssokerperioder(): ArbeidssokerperioderTestdataBuilder {
            return ArbeidssokerperioderTestdataBuilder()
        }
    }

    init {
        arbeidssokerperioder = ArrayList()
    }
}
